front/components/assistant/conversation/actions/VisualizationActionIframe.tsx

```diff
@@ -86,7 +77,7 @@ function useVisualizationDataHandler(
       if (
         !isVisualizationRPCRequest(data) ||
         !isOriginatingFromViz ||
-        data.actionId !== action.id
+        data.index !== visualization.index
```

Comment:
How does this work if you have two visualizations displayed on two different messages? We might have to prefix the index with the agent message to not process events multiple times.

Reply:
You are right, we should prefix